src/unix/fs.c

```diff
@@ -1115,6 +1115,34 @@ int uv_fs_mkdir(uv_loop_t* loop,
 }
 
 
+int uv_fs_mkdirp(uv_loop_t* loop,
+                 uv_fs_t* req,
+                 const char* path,
+                 int mode,
+                 uv_fs_cb cb) {
+  char tmp[256];
+  char *p = NULL;
+  size_t len;
+  int err;
+
+  snprintf(tmp, sizeof(tmp),"%s",path);
+  len = strlen(tmp);
+  if (tmp[len - 1] == '/')
+    tmp[len - 1] = 0;
+
+  for (p = tmp + 1; *p; p++) {
+    if (*p == '/') {
+      *p = 0;
+      err = uv_fs_mkdir(loop, req, tmp, mode, NULL);
```

Comment:
This may potentially execute tons of requests synchronously, which is not acceptable if the user requested async operations.